wandb/agents/pyagent.py

```diff
@@ -228,6 +233,20 @@ def _run_jobs_from_queue(self):
                         )
                         self._exit_flag = True
                         return
+                    if (
+                        self.MAX_FAILURES_AT_START < len(self._errored_runs)
+                        and len(self._errored_runs) >= count
+                    ):
+                        msg = "Detected {} failed runs in a row at start, killing sweep.".format(
+                            self.MAX_FAILURES_AT_START
+                        )
```

Comment:
I'm a bit confused.  It seems this is more like `MAX_FAILURES`.  It seems like this doesn't take into a account whether we are actually at the "start" of a sweep.  How do we want this to be different than FLAPPING_MAX_FAILURES?  The FLAPPING settings actually take into account the notion of "start" via FLAPPING_MAX_SECONDS.  This message also claims these failures are "in a row", the implementation looks like it will reach this case when the total failures over the entire sweep hits this.  It's common in sweeps for arbitrary trials to fail because of OOM or other misconfiguration issues.

Looking at the ticket it seems like FLAPPING_MAX_FAILURES already gives Onfido what they need.  It looks like they just want to stop the sweep of N failures happen in the first X seconds.  I think we should just expose those settings as environment variables.  @raubitsj thoughts?

Reply:
So I was thinking by checking for `len(self._errored_runs) >= count` this will make sure that these failures are happening in a row at the start of the sweep (since `count` will be more than the number of failed runs if this happened midway through the sweep, or if there were successful runs in between).

Reply:
How I understood was that these runs were taking a while before they finally failed. This can be accounted for by increasing the X seconds, but would the users want the same behaviour for sweeps with shorter runs? I was thinking having three 10-min-long runs fail in half an hour vs having three 30-seconds-long runs fail in half an hour might be a little different.

Reply:
Ahh I see, makes sense now.  Since this is specific to WANDB_AGENT, maybe we should just name this env variable `WANDB_AGENT_MAX_FAILURES`.  The name doesn't capture the fact that it's only consecutive failures at start, but we could clearly document it.  `WANDB_AGENT_MAX_CONSECUTIVE_FAILURES_AT_START` doesn't really roll off the tongue 😉

Reply:
or maybe `WANDB_AGENT_MAX_INIT_FAILURES`... @raubitsj any opinions?

Reply:
how about:
WANDB_AGENT_MAX_INITIAL_FAILURES

Reply:
@raubitsj @vanpelt Does this look better?